doc-requirements.txt

```
@@ -1,8 +1,8 @@
 # Requirements for building docs
 # Check these dependencies against doc/conf.py
 -r dev-requirements.txt
-sphinx>=1.0,<3.0
-numpydoc==0.8.0
+sphinx>=1.0
+numpydoc>=0.8.0
```

```suggestion
sphinx>=7.0
numpydoc>=1.6.0
```

I don't think there is any reason that you would need to support older versions for doc building.